ngraph/python/setup.py

```diff
@@ -237,130 +49,126 @@ def cpp_flag(compiler):
+            try:
+                i = build_types.index(str(self.config).lower())
+                self.config = self.cmake_build_types[i]
+            except ValueError:
+                self.announce("Unsupported CMAKE_BUILD_TYPE value: " + self.config, level=4)
+                self.announce("Supported values: {}".format(", ".join(self.cmake_build_types)), level=4)
+                sys.exit(1)
+        super().finalize_options()
+        
+    def run(self):
+        for extension in self.extensions:
+            if extension.name == "_pyngraph":
+                self.build_cmake(extension)
+
+    def build_cmake(self, extension: Extension):
+        """
+        Steps required to build the extension
+        """
+
+        self.announce("Preparing the build environment", level=3)
+
+        build_dir = pathlib.Path(self.build_temp)
+
+        extension_path = pathlib.Path(self.get_ext_fullpath(extension.name))
+
+        os.makedirs(build_dir, exist_ok=True)
+        os.makedirs(extension_path.parent.absolute(), exist_ok=True)
+        
+        # If ngraph_DIR is not set try to build from OpenVINO root
+        root_dir = OPENVINO_ROOT_DIR
+        bin_dir = os.path.join(OPENVINO_ROOT_DIR, "bin")
+        if os.environ.get('ngraph_DIR') is not None:
+            root_dir = PYNGRAPH_ROOT_DIR
+            bin_dir = build_dir
+            
+        self.announce("Configuring cmake project", level=3)
+
+        self.spawn(["cmake", "-H" + root_dir, "-B" + self.build_temp,
+                    "-DCMAKE_BUILD_TYPE={}".format(self.config),
+                    "-DENABLE_CLDNN=OFF",
+                    "-DENABLE_OPENCV=OFF",
+                    "-DENABLE_VPU=OFF",
+                    "-DNGRAPH_PYTHON_BUILD_ENABLE=ON",
+                    "-DNGRAPH_ONNX_IMPORT_ENABLE=ON"])
+        
+        self.announce("Building binaries", level=3)
+        
+        self.spawn(["cmake", "--build", self.build_temp, "--target", extension.name,
+                    "--config", self.config])
+
+        self.announce("Moving built python module to " + str(extension_path), level=3)
+        pyds = [name for name in glob.iglob("{0}/**/{1}*{2}".format(bin_dir,
+                extension.name,
+                sysconfig.get_config_var("EXT_SUFFIX")), recursive=True)]
+        for name in pyds:
+            self.announce("copy " + os.path.join(name), level=3)
+            shutil.copy(name,  extension_path)
+
+
+class InstallCMakeLibs(install_lib):
+
+    def run(self):
+        """
+        Copy libraries from the bin directory and place them as appropriate
+        """
+
+        self.announce("Adding library files", level=3)
+
+        root_dir = os.path.join(OPENVINO_ROOT_DIR, 'bin')
+        if os.environ.get('ngraph_DIR') is not None:
+            root_dir = pathlib.Path(os.environ['ngraph_DIR'])
+
+        lib_ext = ".so"
+        if "linux" in sys.platform or sys.platform == "darwin":
+            lib_ext = ".so"
```

Comment:
Why do we need this `if`? The `lib_ext` already initialised by `.so` value.

Moreover, is this expression correct for macOS? As I know dynamic libraries on macOS have an extension `.dylib`

Reply:
fixed. thanks.